cmd/kubeadm/app/phases/addons/proxy/manifests.go

```diff
@@ -79,6 +79,21 @@ spec:
         - /usr/local/bin/kube-proxy
         - --config=/var/lib/kube-proxy/{{ .ProxyConfigMapKey }}
         - --hostname-override=$(NODE_NAME)
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 10256
+            scheme: HTTP
+          initialDelaySeconds: 5
+          timeoutSeconds: 1
+          successThreshold: 1
+          failureThreshold: 2
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 10256
+            scheme: HTTP
+          timeoutSeconds: 1
```

Comment:
can this fairly low timeout cause flakes?
where can we obtain information about the decisions made here for threshold and timeout values?

also please note that ARM low-spec devices such as the RPI are having trouble in this area ATM and we don't want to block them further.

Reply:
If kubeproxy is taking longer than 1 second to respond there are probably larger issues at play in the cluster, especially since apiserver access is dependent on kube-proxy.

Reply:
@neolit123 Agree in this point with @detiber. But I could increase a bit the value for `readinessProbe` as more critical in the case I try to mitigate/resolve is the minimization of failure time window which basically is related with `livenessProbe`. But I am open to discuss for a solution that applies to the majority of the cases. My main purpose with this PR was to add the probes which make not much sense to me to be absent.

Reply:
i certainly don't want to block this change with regards to the addition of the probes.
but i'm hoping that we don't have to end up mitigating the mitigation eventually, as currently we have zero test signal for ARM low-spec devices.

here is the rather big discussion on RPI liveness probe failures:
https://github.com/kubernetes/kubeadm/issues/413

Reply:
@neolit123 Thanks for all your feedback, is really helpful to understand other perspectives. But based on [this](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes) even if I don't declare it all this will be the default value. (`timeoutSeconds: Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1.`). Another approach in this case could be to not commit the `kubeadm` changes on this commit, and let the community that uses it to decide if they want probes or not and which are the more proper values for them. I added these changes after @MrHohn's advice.  So I will add the release notes you asked for and let me know if you want to remove `kubeadm`'s changes.

Reply:
@stafot 
i'm just raising the concern.
if we are adding the change to `/cluster` we should also add to kubeadm.

Reply:
@neolit123 Your concern is totally reasonable imo.

Reply:
Did a quick check and seems like many other components put `15s` as their timeout:
- https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/kube-apiserver.manifest
- https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/kube-controller-manager.manifest

Not saying those are better, just thinking it maybe safer to start with something looser --- we've seen kube-proxy burning out and failed to response to health check promptly, one specific case is  when too many health checks from LBs are hitting the same health check endpoint.

Reply:
+1

i'd say we add this for both probes as a start:
`initialDelaySeconds: 15` 
`timeoutSeconds: 15`

the delay for the RP can be much lower, but i like to see this in the wild before lowering the values.

Reply:
Agree. Make sense.

Reply:
Agree +1